Cron/PendingOrdersUpdate.php

```diff
@@ -57,6 +65,11 @@ public function execute(): PendingOrdersUpdate
                 $order->setTotalPaid($order->getGrandTotal());
                 $order->addCommentToStatusHistory(__('Successfully completed Amwal payment with transaction ID %1 By Cron Job', $amwalOrderData->getId()));
             }
+
+            if (!$order->hasInvoices()) {
+                $this->logger->error(sprintf('Order %s does not have an invoice', $orderId));
+                $this->invoiceAmwalOrder->execute($order, $amwalOrderData);
+            }
             $this->orderRepository->save($order);
```

Comment:
this needs to move inside the `if ($amwalOrderData && $amwalOrderData['status'] == 'success')` loop and then afterwards also inside the loop, the invoicing code so that an invoice is only is created if an order has been changes